types/pokemon.ts

```diff
@@ -0,0 +1,94 @@
+import { NamedResource } from './common';
+
+export interface Pokemon {
+    id: number;
+    name: string;
+    base_experience: number;
+    height: number;
+    is_default: boolean;
+    order: number;
+    weight: number;
+    moves: PokemonMove[];
+    stats: PokemonStat[];
+    types: PokemonType[];
+}
+
+export interface PokemonMove {
+    move: NamedResource;
+}
+
+export interface PokemonStat {
+    stat: NamedResource;
+    effort: number;
+    base_stat: number;
+}
+
+export interface PokemonType {
+    slot: number;
+    type: NamedResource<
+        | 'normal'
+        | 'dark'
+        | 'fire'
+        | 'water'
+        | 'grass'
+        | 'electric'
+        | 'ice'
+        | 'fighting'
+        | 'poison'
+        | 'ground'
+        | 'flying'
+        | 'psychic'
+        | 'bug'
+        | 'rock'
+        | 'ghost'
+        | 'dragon'
+        | 'steel'
+        | 'fairy'
+        | undefined
+    >;
+}
+
+export interface PokeminListResponse {
+    results: NamedResource[];
+}
+
+export interface FlavorText {
+    flavor_text: string;
+    language: NamedResource;
```

Comment:
Why language prop is using this generic type?

Reply:
The PokéAPI returns almost all resources in the format:
```json
{
    "name": 'Some name',
    "url": 'https://some.url.to.the.resource/'
}
```
And the Pokémon species returned data structure is no exception as the language is a named resource as well as seen here:
```js
{
  "id": 413,
  "name": "wormadam",
  // other properties here
  "flavor_text_entries": [
    {
      "flavor_text": "When the bulb on\nits back grows\nlarge, it appears\fto lose the\nability to stand\non its hind legs.",
      "language": {
        "name": "en", // NamedResource shape
        "url": "https://pokeapi.co/api/v2/language/9/"
      },
      "version": {
        "name": "red",
        "url": "https://pokeapi.co/api/v2/version/1/"
      }
    }
  ],
// other properties here
}
```
So I decided to create that NamedResource interface in order to not repeat the same definitions over and over